lite2/client.go

```diff
@@ -649,15 +650,34 @@ func (c *Client) cleanup(stopHeight int64) error {
 		return errors.Wrap(err, "can't get last trusted height")
 	}
 
-	// 3) Remove all headers and validator sets.
-	if stopHeight < oldestHeight {
-		stopHeight = oldestHeight
+	// 3) Ensure the start height and stop height are within the range of headers stored.
+	if stopHeight == 0 || stopHeight > latestHeight {
+		stopHeight = latestHeight
+	}
+	if startHeight < oldestHeight {
```

Comment:
what about startHeight == 0?

Reply:
if startHeight is 0 than implicitly it is less than oldestHeight